packages/store/src/create-server-workspace.ts

```diff
@@ -0,0 +1,347 @@
+  const result: OpenAPIV3_1.PathsObject = {}
+
+  for (const [path, methods] of Object.entries(paths)) {
+    if (!methods) {
+      continue
+    }
+
+    const filteredMethods: Record<string, any> = {}
+
+    for (const [method, operation] of Object.entries(methods)) {
+      if (httpMethods.has(method.toLowerCase())) {
+        filteredMethods[method] = operation
+      }
+    }
+
+    if (Object.keys(filteredMethods).length > 0) {
+      result[path] = filteredMethods
+    }
+  }
+
+  return result
+}
+
+/**
+ * Escapes path keys in an OpenAPI PathsObject to be JSON Pointer compatible.
+ * This is necessary because OpenAPI paths can contain characters that need to be escaped
+ * when used as JSON Pointer references (like '/' and '~').
+ *
+ * @example
+ * Input: { "/users/{id}": { ... } }
+ * Output: { "/users~1{id}": { ... } }
+ */
+export function escapePaths(paths: OpenAPIV3_1.PathsObject) {
+  const result: OpenAPIV3_1.PathsObject = {}
+  Object.keys(paths).forEach((path) => {
+    result[escapeJsonPointer(path)] = paths[path]
+  })
+
+  return result
+}
+
+/**
+ * Externalizes components by turning them into refs.
+ */
+export function externalizeComponentReferences(
+  document: OpenAPIV3_1.Document,
+  meta: { mode: 'ssr'; name: string; baseUrl: string } | { mode: 'static'; name: string; directory: string },
+) {
+  const result: Record<string, any> = {}
+
+  if (!document.components) {
+    return result
+  }
+
+  Object.entries(document.components).forEach(([type, component]) => {
+    result[type] = {}
+    Object.keys(component).forEach((name) => {
+      const ref =
+        meta.mode === 'ssr'
+          ? `https://${meta.baseUrl}#/${meta.name}/components/${type}/${name}`
+          : `${meta.directory}/chunks/${meta.name}/components/${type}/${name}.json/#`
+
+      result[type][name] = { '$ref': ref }
+    })
+  })
+
+  return result
+}
+
+/**
+ * Externalizes paths operations by turning them into refs.
+ */
+export function externalizePathReferences(
+  document: OpenAPIV3_1.Document,
+  meta: { mode: 'ssr'; name: string; baseUrl: string } | { mode: 'static'; name: string; directory: string },
+) {
+  const result: Record<string, any> = {}
+
+  if (!document.paths) {
+    return result
+  }
+
+  Object.entries(document.paths).forEach(([path, pathItem]) => {
+    if (!pathItem) {
+      return result
+    }
+
+    result[path] = {}
+
+    const escapedPath = escapeJsonPointer(path)
+
+    Object.keys(pathItem).forEach((type) => {
+      if (httpMethods.has(type)) {
+        const ref =
+          meta.mode === 'ssr'
+            ? `https://${meta.baseUrl}#/${meta.name}/operations/${escapedPath}/${type}`
+            : `${meta.directory}/chunks/${meta.name}/operations/${escapedPath}/${type}.json/#`
+
+        result[path][type] = { '$ref': ref }
+      }
+    })
+  })
+
+  return result
+}
+
+const DEFAULT_ASSETS_FOLDER = 'assets'
+const WORKSPACE_FILE_NAME = 'scalar-workspace.json'
```

Comment:
Personally, I like to put const’s like this at the top of the file, but that might be just me